admin-dev/themes/new-theme/js/app/utils/init-components.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import TranslatableField from '@js/components/translatable-field.js';
+import TranslatableInput from '@js/components/translatable-input.js';
+import TinyMCEEditor from '@js/components/tinymce-editor.js';
+
+const initComponents = () => {
+  // Initialize the prestashop global object
+  window.prestashop = {
+    components: {
+      translatableInput: {
+        enabled: false,
+        enable() {
+          // Already enabled
+          if (window.prestashop.components.translatableInput.enabled === true) {
+            return;
+          }
+          window.prestashop.components.translatableInput.instance = new TranslatableInput();
+          window.prestashop.components.translatableInput.enabled = true;
+        },
+        instance: null,
```

Comment:
What troubles me with this way of creating the component is that all the internal variables are directly accessible via `window.prestashop.components.translatableInput.enabled` and could be messed with

Ideally they should be private, but javascript doesn't manage this but at least if we had some modules that only expose the required api it would make it more reliable

Reply:
> and could be messed with

What are you afraid of ? 🤔  modules messing with each other, or deeper, security issues ?